packages/admin/src/pages/super-admin-dashboard/user/[id]/change-department.test.tsx

```diff
@@ -0,0 +1,144 @@
+    description: 'this is a description',
+    label: 'Find',
+  },
+  {
+    id: '2',
+    name: 'APPLICANT',
+    description: 'this is another description',
+    label: 'Applicant',
+  },
+  {
+    id: '3',
+    name: 'ADMIN',
+    description: 'this is another description',
+    label: 'Administrator',
+  },
+  {
+    id: '4',
+    name: 'SUPER_ADMIN',
+    description: 'this is a super admin description',
+    label: 'Super administrator',
+  },
+];
+
+const getMockDepartment = (): Department => ({
+  id: '1',
+  name: 'Test Department',
+});
+
+const getMockUser = (): User => ({
+  gapUserId: 'mockId',
+  sub: 'sub',
+  colaSub: 'colaSub',
+  emailAddress: 'test.superadmin@gmail.com',
+  roles: [getMockRoles()[0], getMockRoles()[3]],
+  created: 'NULL',
+});
+
+describe('Change department page', () => {
+  beforeEach(jest.clearAllMocks);
+  describe('UserPage component', () => {
+    test('Should render the correct content', () => {
+      const component = (
+        <UserPage
+          csrfToken="csrf"
+          pageData={{
+            user: getMockUser(),
+            departments: [getMockDepartment()],
+          }}
+          formAction="."
+          fieldErrors={[]}
+          previousValues={{ department: '1' }}
+        />
+      );
+      render(component);
+      expect(
+        screen.getByRole('heading', { name: "Change the user's department" })
+      ).toBeVisible();
+      expect(screen.getByText('Test Department')).toBeVisible();
+      expect(
+        screen.getByText('Test Department').previousSibling
+      ).toHaveAttribute('type', 'radio');
+      expect(screen.getByText('Change department')).toBeVisible();
+      expect(screen.getByText('Manage departments')).toBeVisible();
+    });
+
+    test('Should throw an error if no department is selected', async () => {
+      const component = (
+        <UserPage
+          csrfToken="csrf"
+          pageData={{
+            user: getMockUser(),
+            departments: [getMockDepartment()],
+          }}
+          formAction="."
+          fieldErrors={[
+            { fieldName: 'department', errorMessage: 'Select a department' },
+          ]}
+          previousValues={{ department: '1' }}
+        />
+      );
+      render(component);
+      expect(screen.getByText('There is a problem')).toBeVisible();
+      expect(
+        screen.getByText('There is a problem').nextSibling
+      ).toHaveTextContent('Select a department');
+    });
+  });
+
+  describe('getServerSideProps', () => {
+    const mockGetChangeDepartmentPage = jest.mocked(getChangeDepartmentPage);
+    const mockParseBody = jest.mocked(parseBody);
+    test('Should redirect to user page if department is selected', async () => {
+      mockGetChangeDepartmentPage.mockReturnValueOnce(
+        Promise.resolve({
+          user: getMockUser(),
+          departments: [{ id: '4', name: 'hello world' }],
+        })
+      );
+      mockParseBody.mockResolvedValue({ department: 'fake department' });
```

Comment:
I should have cleaned this up before I left the call, but the best way to do this is probably


```suggestion
      mockGetChangeDepartmentPage.mockResolvedValueOnce(
        {
          user: getMockUser(),
          departments: [{ id: '4', name: 'hello world' }],
        }
      );
      mockParseBody.mockResolvedValueOnce({ department: 'fake department' });
```

then you don't need the `Promise.resolve` invocation anywhere, and because we're only mocking once this mock implementation won't 'bleed' into other tests.